pkg/resources/schema.go

```diff
@@ -315,18 +405,30 @@ func UpdateContextSchema(ctx context.Context, d *schema.ResourceData, meta any)
 	}
 
 	if d.HasChange("with_managed_access") {
-		var err error
-		if _, ok := d.GetOk("with_managed_access"); ok {
-			err = client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				EnableManagedAccess: sdk.Pointer(true),
-			})
-		} else {
-			err = client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				DisableManagedAccess: sdk.Pointer(true),
-			})
+		if v := d.Get("with_managed_access").(string); v != BooleanDefault {
+			var err error
+			parsed, err := booleanStringToBool(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			if parsed {
+				err = client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
+					EnableManagedAccess: sdk.Pointer(true),
+				})
+			} else {
+				err = client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
+					DisableManagedAccess: sdk.Pointer(true),
+				})
+			}
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("error handling with_managed_access on %v err = %w", d.Id(), err))
+			}
 		}
-		if err != nil {
-			return diag.FromErr(fmt.Errorf("error changing management state on %v err = %w", d.Id(), err))
+		// managed access can not be UNSET to a default value
+		if err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
```

Comment:
I don't get this part; we always disable managed access even if we enabled it a second ago in line 416. I think this can cause infinite plan for with_managed_access field.

Reply:
Fixed.